src/Core/Listener/EmailConfirmationMailer.php

```diff
@@ -11,20 +11,22 @@
 
 namespace Flarum\Core\Listener;
 
-use Flarum\Core;
 use Flarum\Core\EmailToken;
+use Flarum\Core\Jobs\MailJob;
+use Flarum\Core\Support\DispatchJobsTrait;
 use Flarum\Core\User;
 use Flarum\Event\UserEmailChangeWasRequested;
 use Flarum\Event\UserWasRegistered;
 use Flarum\Forum\UrlGenerator;
 use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Bus\Dispatcher as JobDispatcher;
 use Illuminate\Contracts\Events\Dispatcher;
 use Illuminate\Contracts\Mail\Mailer;
-use Illuminate\Mail\Message;
 use Symfony\Component\Translation\TranslatorInterface;
 
 class EmailConfirmationMailer
 {
+    use DispatchJobsTrait;
```

Comment:
Honestly, I don't think we need this trait, if all it does is shortening `$this->queue->dispatch()` to `$this->dispatch()`, at the cost of one more step to find out where this method is coming from, and why we need to inject the seemingly-unused `$queue` attribute. :)